fix(compiler-cli): handle conditional expressions when extracting dependencies

clydin opened the pull request:
Updates the HMR dependencies extraction logic to handle conditional expressions. For example, `providers: [condition ? providersA : providersB]`.

Closes #59634

clydin committed:
fix(compiler-cli): handle conditional expressions when extracting dependencies

Updates the HMR dependencies extraction logic to handle conditional expressions.
For example, `providers: [condition ? providersA : providersB]`.

crisbeto commented on packages/compiler-cli/src/ngtsc/hmr/src/extract_dependencies.ts:
```diff
@@ -300,6 +300,10 @@ class PotentialTopLevelReadsVisitor extends o.RecursiveAstVisitor {
       return (parent.propertyName || parent.name) === node;
     }
 
+    if (ts.isConditionalExpression(parent)) {
```
For the `new` expressions we should be able to do `ts.isNewExpression(parent) && parent.expression === node`.

clydin replied:
Added a separate commit for handling new expressions.

clydin committed:
fix(compiler-cli): handle new expressions when extracting dependencies

Updates the HMR dependencies extraction logic to handle new expressions.
For example, `deps: [[new Optional(), dep]]`.

AndrewKushnir commented:
This PR was merged into the repository by commit 5a13dff22c227e731b46171a2dde3686a50c5c32.

The changes were merged into the following branches: main, 19.1.x
